packages/builder/src/pages/builder/app/[application]/data/datasource/[datasourceId]/index.svelte

```diff
@@ -12,16 +12,25 @@
   import PromptQueryModal from "./_components/PromptQueryModal.svelte"
   import SettingsPanel from "./_components/panels/Settings.svelte"
   import { helpers } from "@budibase/shared-core"
+  import { goto } from "@roxi/routify"
 
   let selectedPanel = null
   let panelOptions = []
 
   // datasources.selected can return null temporarily on datasource deletion
-  $: datasource = $datasources.selected || {}
+  $: datasource = $datasources.selected
+  $: {
+    if (!datasource) {
+      $goto("./datasource")
+    }
+  }
```

Comment:
Ooh I didn't know you could do this! I assumed reactive statements had to be variable assignments. Cool!

Reply:
You can indeed! You can have any arbitrary code inside a reactive statement like this, and svelte re-run it when any of the variables used inside the statement change. What svelte determines a "change" worth invalidating reactive statements is either a reassignment (if a normal JS variable) or a store value change (if using a svelte store).